importer/helper.go

```diff
@@ -50,12 +50,29 @@ type ImportHelper struct {
 	Log *log.Logger
 }
 
+func readGenesis(in io.Reader) (bookkeeping.Genesis, error) {
+	var genesis bookkeeping.Genesis
+	gbytes, err := ioutil.ReadAll(in)
+	if err != nil {
+		return bookkeeping.Genesis{}, fmt.Errorf("error reading genesis, %v", err)
+	}
+	err = protocol.DecodeJSON(gbytes, &genesis)
+	if err != nil {
+		return bookkeeping.Genesis{}, fmt.Errorf("error decoding genesis, %v", err)
+	}
+	return genesis, nil
+}
+
 // Import is the main ImportHelper function that glues together a directory full of block files and an Importer objects.
 func (h *ImportHelper) Import(db idb.IndexerDb, args []string) {
 	// Initial import if needed.
 	genesisReader := GetGenesisFile(h.GenesisJSONPath, nil, h.Log)
-	_, err := EnsureInitialImport(db, genesisReader, h.Log)
+	genesis, err := readGenesis(genesisReader)
+	maybeFail(err, h.Log, "readGenesis() error")
```

Comment:
Still kind of a mess, I didn't refactor this to use the bookkeeping.Genesis file because it's the deprecated `import` command...